arrow-core-test/src/main/kotlin/arrow/test/laws/FoldableLaws.kt

```diff
@@ -154,7 +185,86 @@ object FoldableLaws {
 
   fun <F> Foldable<F>.`toList turn items into a list`(G: Gen<Kind<F, Int>>, EQ: Eq<Int>) =
     forAll(G) { fa: Kind<F, Int> ->
+      val result = fa.toList()
       val expected = fa.foldRight(Eval.now(emptyList<Int>())) { v, acc -> acc.map { listOf(v) + it } }.value()
-      fa.toList().size.equalUnderTheLaw(expected.size, EQ)
+      result.eqK(expected, EQ)
+    }
+
+  fun <F> Foldable<F>.`fold should combine all items`(G: Gen<Kind<F, Int>>, EQ: Eq<Int>) =
+    forAll(G) { fa: Kind<F, Int> ->
+      with(Int.monoid()) {
+        fa.fold(this).equalUnderTheLaw(fa.foldLeft(empty()) { acc, a -> acc.combine(a) }, EQ)
+      }
+    }
```

Comment:
Please use a non-commutative monoid here. That'd verify the correct order as well. Otherwise one could implement `fold` reversing the order which would never show with this test.